local-volume/provisioner/pkg/deleter/deleter.go

```diff
@@ -82,75 +98,110 @@ func (d *Deleter) deletePV(pv *v1.PersistentVolume) error {
 
 	// Default is filesystem mode, so even if volume mode is not specified, mode should be filesystem.
 	volMode := v1.PersistentVolumeFilesystem
+	runjob := false
 	if pv.Spec.VolumeMode != nil && *pv.Spec.VolumeMode == v1.PersistentVolumeBlock {
 		volMode = v1.PersistentVolumeBlock
+		runjob = d.RuntimeConfig.UseJobForCleaning
 	}
 
-	if d.ProcTable.IsRunning(pv.Name) {
-		// Run in progress, nothing to do,
+	// Exit if cleaning is still in progress.
+	if d.CleanupStatus.InProgress(pv.Name, runjob) {
 		return nil
 	}
 
-	err = d.ProcTable.MarkRunning(pv.Name)
+	// Check if cleaning was just completed.
+	state, err := d.CleanupStatus.RemoveStatus(pv.Name, runjob)
 	if err != nil {
 		return err
 	}
 
-	go d.asyncDeletePV(pv, volMode, mountPath, config)
+	if state == CSSucceeded {
```

Comment:
Can a switch statement be used here instead?

Reply:
Yes, I was on the fence with my choice there. Fixed now.